### DISCUSSION TITLE
Publishing your website live using Git, your approaches

### DISCUSSION THREAD

**Discussion by vandenbroucke** [GetPublii/Publii#1342] [General] [answered]
Hello everyone,

I'm loving the fact that you can now publish your site using Git.
I have setup Github actions to automatically deploy the files to an azure static website. 
The publish however overwrites the whole repository, which forces me to have a repo for public site and repo for all the publii code.

I feel that it would be better having an option to publish to a publish folder (to git) while keeping all the other source code. 
In my opinion it would be great to have a client sync up their site from the git repo, make changes and publish it.
While I'm still able to work on the technical parts of the publii site.

How do you work and or publish with clients that want to edit?

**Comment by dziudek** [comment]
Hi,

I do not understand your case - Publii pushes to the repository only output files. At this moment we have no plans to put on the repository input files - there is many potential issues with it. Git sync is an deployment option, not the backup option.

**Reply by vandenbroucke** [reply]
Now for example I would use a Github Workflow which resides inside the repo to deploy to Azure. However, since Publii overwrites the WHOLE repo, the workflows are removed.

**Reply by dziudek** [reply, marked as answer]
@vandenbroucke - if you have any files which should be placed on the repository, then you should place them under /input/root-files/ - then it will be uploaded/updated with your deployments

**Reply by vandenbroucke** [reply]
@dziudek Thank you very much, I've placed my workflow files there. The combination with an Azure Static Web app deployment is effortless!
